File: src/sequence/dialogitem.cpp
#include "dialogitem.h"

namespace ks {

DialogItem::DialogItem(const bn::string<16>& title, const bn::string<512>& message)
    : _title(title), _message(message) {}

SequenceType DialogItem::type() const {
    return ks::SequenceType::Dialog;
}

SequenceItem* DialogItem::clone() const {
    return new DialogItem(*this);
}

const bn::string<16>& DialogItem::title() const {
    return _title;
}

const bn::string<512>& DialogItem::message() const {
    return _message;
}

bool DialogItem::finished() const {
    return _finished;
}

void DialogItem::finished(bool flag) {
    _finished = flag;
}

} // namespace ks

File: tools/converter/src/dto/dialog_item.py
from src.dto.sequence_item import SequenceItem, SequenceType


class DialogItem(SequenceItem):
    def some_method(self):
        pass

    def __init__(self, identifier: str, actor: str, message: str, actor_ref: str = None, label_name: str = None):
        super().__init__(SequenceType.DIALOG)
        self.id = identifier
        self.actor = actor
        self.actor_ref = actor_ref
        self.message = message
        self.label_name = label_name

    def __str__(self):
        return f"DialogItem(actor={self.actor}, actor_ref={self.actor_ref}, message={self.message}, id={self.id})"
